app/src/main/java/org/mozilla/focus/tiles/HomeTilesManager.kt

```diff
@@ -18,6 +18,19 @@ const val KEY_TITLE = "title"
 const val KEY_IMG = "img"
 const val KEY_ID = "identifier"
 
+data class HomeTile (
+        val url: String,
+        val title: String,
+        val imagePath: String?,
+        // unique id used to identify specific home tiles, e.g. for deletion, etc.
+        val id: String
+) {
+    constructor(jsonObject: JSONObject): this(jsonObject.getString(KEY_URL),
```

Comment:
nit: Since this is just a data container and its constructor is naturally used to fill its data, I think a more natural way of constructing this is statically:
```kotlin
data class HomeTile(
    ...
) {
    companion object {
        fun fromJSONObject(jsonObj: JSONObject): HomeTile? {
            return HomeTile(...)
        }
    }
}
```

Benefits:
- Can return null (e.g. if the JSONObject doesn't have the fields you expect/require)
- Defining multiple static constructors is more natural than having many constructors
- iirc, `this` constructor must be called before you can execute code (e.g. store local variables) which are helpful to make the code more readable.